From 3dd9ca403c12e16b5d7d263a390d43330b4dca8a Mon Sep 17 00:00:00 2001
From: Jeff Hammond <jeff.science@gmail.com>
Date: Fri, 21 Feb 2014 13:09:41 -0600
Subject: [PATCH 1/2] add CVAR to disable Abort message

PETSc users are confused by error messages because they can't be
troubled to understand them.

Barry Smith requested that MPICH not indicate that it is aborting.

this is part 1 of N patches to address this.
---
 src/mpi/init/abort.c | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/src/mpi/init/abort.c b/src/mpi/init/abort.c
index 13a33d8e..24de3544 100644
--- a/src/mpi/init/abort.c
+++ b/src/mpi/init/abort.c
@@ -10,6 +10,22 @@
 #include <stdlib.h>
 #endif
 
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_SUPPRESS_ABORT_MESSAGE
+      category    : ERROR_HANDLING
+      type        : boolean
+      default     : true
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_ALL_EQ
+      description : Disable printing of abort error message.
+
+=== END_MPI_T_CVAR_INFO_BLOCK ===
+*/
+
 /* -- Begin Profiling Symbol Block for routine MPI_Abort */
 #if defined(HAVE_PRAGMA_WEAK)
 #pragma weak MPI_Abort = PMPI_Abort
@@ -114,7 +130,8 @@ int MPI_Abort(MPI_Comm comm, int errorcode)
 	MPIU_Snprintf(comm_name, MPI_MAX_OBJECT_NAME, "comm=0x%X", comm);
     }
     /* FIXME: This is not internationalized */
-    MPIU_Snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name, errorcode, comm_ptr->rank);
+    if (!MPIR_CVAR_SUPPRESS_ABORT_MESSAGE)
+        MPIU_Snprintf(abort_str, 100, "application called MPI_Abort(%s, %d) - process %d", comm_name, errorcode, comm_ptr->rank);
     mpi_errno = MPID_Abort( comm_ptr, mpi_errno, errorcode, abort_str );
     /* --BEGIN ERROR HANDLING-- */
     if (mpi_errno != MPI_SUCCESS) goto fn_fail;
-- 
1.8.5.3

